app/javascript/react/screens/App/Overview/OverviewActions.js

```diff
@@ -245,31 +245,41 @@ export const archiveTransformationPlanAction = (url, id) => {
   return _archiveTransformationPlanActionCreator(uri.toString());
 };
 
-export const toggleScheduleMigrationModal = planId => ({
+export const toggleScheduleMigrationModal = plan => ({
   type: V2V_TOGGLE_SCHEDULE_MIGRATION_MODAL,
-  payload: planId
+  payload: plan
 });
 
 export const scheduleMigration = payload => dispatch =>
   dispatch({
     type: V2V_SCHEDULE_MIGRATION,
     payload: new Promise((resolve, reject) => {
-      let url = `/api/service_templates/${payload.planId}`;
+      const {
+        scheduleTime,
+        plan: { id: planId }
+      } = payload;
+      const scheduleId = (payload.plan.schedules && payload.plan.schedules[0].id) || null;
```

Comment:
This will be a `TypeError` if `payload.plan.schedules` is an empty array (truthy but `[0]` is undefined). Not sure if we'll ever hit that case, but we should maybe do:

```
const firstSchedule = payload.plan.schedules && payload.plan.schedules[0];
const scheduleId = firstSchedule ? firstSchedule.id : null;
```

Reply:
if it exists, as in if ` payload.plan.schedules` passes, there will be something inside 👍